### DISCUSSION TITLE
Why do some changes do not happen automatically, but require the user to press apply?

### DISCUSSION THREAD

**Discussion by nNoidea** [darkreader/darkreader#8244] [Q&A]
So I find a easy and short solution to fix the transparency issue on my universities website, however after restarting the website, my changes I made in the dev tools do not apply and I need to press on the Dark Reader's dev tool's "apply" button to work.

Is this caused due to me using a id searcher based on a string?
Here's the change I made:
```
ufora.ugent.be

CSS
div[id^=d2l-]{
background-color: #181a1b;
}

```
The site is constantly changing the ids of some elements that appear to be transparent, however they always start with the "d2l-" part, so I just went for that and **everything** that was appearing wrong (transparent) was just looking fine.

However I need to manually press the apply button after each refresh. **How can I make dark reader apply this automatically?** I tried putting a transition delay, but it did nothing. 😢

Here are some pics, to show what I mean:
1. Opening the website for the first time or after refreshing the website (while the changes are already saved in Dark Reader):
![image](https://user-images.githubusercontent.com/58337790/154898081-4d1f7e38-077d-434a-9dc4-63dbfe8d6fa8.png)

2. After pressing the apply button manually:
![image](https://user-images.githubusercontent.com/58337790/154898085-5d37ef81-63a9-4c5a-9928-9a51a259c46a.png)

**Comment by Gusted** [comment]
Try:
```css
CSS
div[id^="d2l-"] {
    background-color: var(--darkreader-neutral-dark) !important;
}
```

**Reply by nNoidea** [reply]
Nope does not work. Everything that was transparent, stays transparent.

Here's the style info after your change, I don't think it even changed anything and I pressed on apply to be sure:
![image](https://user-images.githubusercontent.com/58337790/155149748-a7225d1e-2cc7-46ee-8c32-6aabb16195c4.png)
The class id always starts with "d2l-"
![image](https://user-images.githubusercontent.com/58337790/155149890-db5b3927-0f52-4e8a-a1f8-9dea6eb07e99.png)

**Reply by nNoidea** [reply]
Anything else I can try? Is this a bug?

**Reply by Gusted** [reply]
What about:

```CSS
CSS
div[id^="d2l"],
div[class^="d2l"] {
    background-color: var(--darkreader-neutral-dark) !important;
}
```

**Reply by nNoidea** [reply]
If I press the apply button manually it does make stuff darker (`class` part kinda makes some stuff that aren't supposed to be dark, darker. And I had to make the `var(--darkreader-neutral-dark)` directly a color value to see the changs, just `var(--darkreader-neutral-dark)` as the value changes nothing).

However when I completely turn off my browser and reopen it and go to the website without touching the dev tools (the changes/settings are saved tho), it does nothing to the site again. :(

**Reply by Gusted** [reply]
> However when I completely turn off my browser and reopen it and go to the website without touching the dev tools (the changes are there tho), it does nothing to the site again. :(

Could find the `<style class="darkreader darkreader--override" media="screen">` and see if the fixes are in their?

**Reply by nNoidea** [reply]
> Could find the <style class="darkreader darkreader--override" media="screen"> and see if the fixes are in their?

It has `fallback` instead of `override`.
![image](https://user-images.githubusercontent.com/58337790/156387518-adee63ce-9320-4273-83fe-79cc4b0535a2.png)


When I press manually apply it still says `fallback`.
![image](https://user-images.githubusercontent.com/58337790/156387735-f12a4c16-d44f-46bd-9dd3-1b46910c02ac.png)

**Reply by Gusted** [reply]
The fallback isn't the style that contains these fixes, it's the `darkreader--override`

**Reply by nNoidea** [reply]
When I press the apply button manually it does add it, but if I don't I couldn't find it.
![image](https://user-images.githubusercontent.com/58337790/156388855-676cdfa9-691d-49a6-9775-c86bfccafb04.png)

1 weird thing maybe, after I press "apply" and don't complete shutdown all the tabs, but shutdown all the subject websites and open the subject website again. It does apply the new settings automatically. Just not when I completely shutdown my browser. This only happens for this specific search setting and not if I had to change 1 random thing based on id or class without needing to search.

After I completely shutdown and restart chrome:
![image](https://user-images.githubusercontent.com/58337790/156389879-6f10da66-30c4-41cc-988e-6daf67b66300.png)

**Reply by Gusted** [reply]
Hmm, that's weird so `--override` gets deleted? Never seen that behavior before...

**Reply by nNoidea** [reply]
> Hmm, that's weird so `--override` gets deleted? Never seen that behavior before...

I am not sure if it gets deleted, but it just does not add (below code), until I press apply :(
```
div[id^="d2l"]{
    background-color: var(--darkreader-neutral-dark) !important;
}
```

It does add other stuff that aren't just filters but actual IDs or classes

**Reply by Gusted** [reply]
Check if there are any errors in the background-page of the extension, it's most likely that the fix is not getting saved somehow.

**Reply by EnziTheViking** [reply]
I know this is old, but I could not find someone else who had this issue. The solution is to add "!important"


================================
ufora.ugent.be

CSS
div[id^=d2l-]{
	background-color: #181a1b !important;
}

================================

**Comment by Zer0zzy** [comment]
There are two main issues I've found. All settings and menu references are for the new version of the extension.

1. Make sure you have !important at the end of your line, as seen in other replies here.
2. Make sure the "Immediate Modify" setting is disabled. Find it under (See All Options > Font & More > Immediate Modify) and set it to "No". 

I have not found any documentation yet that talks about why you would want immediateModify turned on.
